core/net/mac/tsch/tsch-adaptive-timesync.c

```diff
@@ -86,12 +86,16 @@ timesync_entry_add(int32_t val, uint32_t time_delta)
 static void
 timesync_learn_drift_ticks(uint32_t time_delta_asn, int32_t drift_ticks)
 {
-  /* should fit in 32-bit unsigned integer */
-  uint32_t time_delta_ticks = time_delta_asn * tsch_timing[tsch_ts_timeslot_length];
+  /* should fit in a 32-bit integer */
+  int32_t time_delta_ticks = time_delta_asn * tsch_timing[tsch_ts_timeslot_length];
   int32_t real_drift_ticks = drift_ticks + compensated_ticks;
   int32_t last_drift_ppm = (int32_t)((int64_t)real_drift_ticks * TSCH_DRIFT_UNIT / time_delta_ticks);
 
   drift_ppm = timesync_entry_add(last_drift_ppm, time_delta_ticks);
+
+  TSCH_LOG_ADD(tsch_log_message,
+      snprintf(log->message, sizeof(log->message),
+          "drift %ld", drift_ppm / 256));
```

Comment:
But wait, this is also only done when receiving from the time source isn't it?
A good thing with the Rx/Tx logs is that they don't suffer from the 48 byte limitation as they report the numbers via a dedicated struct, so I feel it would be convenient so simply add a field for ppms next to estimated_drift.

But I'm also fine if you don't want to change this.